alternator/executor.cc

```diff
@@ -1636,19 +1649,56 @@ future<executor::request_return_type> rmw_operation::execute(service::storage_pr
     });
 }
 
+bytes query_cache_key(const std::string_view& query, char query_type) {
+   auto b = md5_hasher::calculate(query);
+   b[0] = query_type; // this avoids key collision for different expression types
+   return b;
+}
+
+template<typename T, typename F>
+requires requires(F f, std::string_view v) {
+    { f(v) } -> std::same_as<T>;
+}
+static T parse_or_get_from_cache(lw_shared_ptr<expression_cache_t> cache, F parse_func, std::string_view query, char query_type) {
```

Comment:
Why pass the cache as lw_shared_ptr? the cache lifetime should be fully contained within the executor's.